core/services/workflows/models.go

```diff
@@ -106,17 +106,42 @@ func (w *workflow) dependents(start string) ([]*step, error) {
 	return steps, nil
 }
 
-// step wraps a stepDefinition with additional context for dependencies and execution
+// step wraps a Vertex with additional context for dependencies and execution
 type step struct {
-	stepDefinition
-	dependencies      []string
+	Vertex
 	capability        capabilities.CallbackCapability
 	config            *values.Map
 	executionStrategy executionStrategy
 }
 
+type Vertex struct {
+	StepDefinition
+	dependencies []string
+}
+
+// StaticWorkflow is an intermediate representation of a workflow wherein all the graph
+// vertices are represented and validated. The representation is static and cannot be executed.
+type StaticWorkflow struct {
+	ID string
+	graph.Graph[string, *Vertex]
+
+	Triggers []*StepDefinition
+
+	Spec *WorkflowSpec
+}
+
+func (w *StaticWorkflow) Hasher() func(s *Vertex) string {
```

Comment:
What's the purpose behind this change?

Reply:
the graph lib takes a func that compute the id of a vertex. that lib calls it a hash, and the existing code call it stepHash, and i just reused the name.

the computation need to be common btw the intermediate rep and the workflow so that it's possible to copy the graph structure.